code/game/turfs/simulated/walls.dm

```diff
@@ -46,6 +46,10 @@
 			/obj/structure/window/full,
 			/obj/structure/window/full/phoron
 		)
+		if(istype(src, /turf/simulated/wall/r_wall))
+			LAZYADD(canSmoothWith, /turf/simulated/wall)
+		if(src.type == /turf/simulated/wall)
+			LAZYADD(canSmoothWith, /turf/simulated/wall/r_wall)
```

Comment:
this should probably be a proc on `/turf/simulated/wall`. something like `/turf/simulated/wall/proc/select_wall_smooth()` which gets overridden to add the other wall type. doing istype(src) is never good